### PR TITLE
Update F2B wrapper to show possible errors with IPTables

### LINKED ISSUE #2169: [BUG] ./setup.sh debug fail2ban does not show banned IPs (RaspberryPi)
./setup.sh debug fail2ban does not show banned IPs for some reason
```
root@raspberrypi:/etc/mailserver# ./setup.sh debug fail2ban
No IPs have been banned.
```
However, there are banned IPs actually.
```
root@raspberrypi:/etc/mailserver# docker exec -it mailserver fail2ban-client status postfix
Status for the jail: postfix
|- Filter
|  |- Currently failed:	0
|  |- Total failed:	0
|  `- File list:	/var/log/mail.log
`- Actions
   |- Currently banned:	1
   |- Total banned:	1
   `- Banned IP list:	167.***.***.203
   ```
When I'm trying to verify iptables rules form inside the container I get the following error
```
root@mail:/# iptables -S
iptables v1.8.2 (legacy): can't initialize iptables table `filter': Table does not exist (do you need to insmod?)
Perhaps iptables or your kernel needs to be upgraded.
```
However, on my host system iptables works perfectly.
RaspberryPi is running on latest Raspbian 10 Buster OS.

My docker-compose.yml
```version: '3.8'
services:
  mailserver:
    container_name: mailserver
    image: docker.io/mailserver/docker-mailserver
    hostname: mail
    domainname: domain.com
    environment:
      - ENABLE_FAIL2BAN=1
      - ENABLE_POP3=1
      - ENABLE_SPAMASSASSIN=1
      - SPAMASSASSIN_SPAM_TO_INBOX=1
      - ENABLE_POSTGREY=1
      - ENABLE_SASLAUTHD=0
      - ONE_DIR=1
      - SSL_TYPE=manual
      - SSL_CERT_PATH=/etc/letsencrypt/live/domain.com/fullchain.pem
      - SSL_KEY_PATH=/etc/letsencrypt/live/domain.com/privkey.pem
    env_file:
      - mailserver.env
    ports:
      - "25:25"
      - "110:110"
      - "143:143"
      - "465:465"
      - "587:587"
      - "993:993"
      - "995:995"
    volumes:
      - ./config/data/maildata:/var/mail
      - ./config/data/mailstate:/var/mail-state
      - ./config/data/maillogs:/var/log/mail
      - ./config:/tmp/docker-mailserver
      - /etc/letsencrypt:/etc/letsencrypt:ro
      - /etc/localtime:/etc/localtime:ro
    cap_add:
      - NET_ADMIN
      - SYS_PTRACE
    restart: unless-stopped

### PR DESCRIPTION
**Pull request by georglauterbach**
# Description

The `fail2ban` wrapper would show `No IPs have been banned` even when IPTables did not work. This is now shown and the script aborts.

Partly helps with #2169

## Type of change

- [x] Improvement (non-breaking change that does improve existing functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (README.md or the documentation under `docs/`)
- [x] If necessary I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: target/bin/fail2ban [hunk 1]
```diff
@@ -3,6 +3,23 @@
 # shellcheck source=../scripts/helper-functions.sh
 . /usr/local/bin/helper-functions.sh
 
+if ! iptables -L &>/tmp/iptables-check-command-output
+then
+  echo "IPTables does not seem to work properly. The output of 'iptables -L' was:
+
+$(</tmp/iptables-check-command-output)
```
Comment (casperklein) [inline target/bin/fail2ban:10]:
```suggestion
if ! IPTABLES_OUTPUT=$(iptables -L 2>&1)
then
  echo "IPTables does not seem to work properly. The output of 'iptables -L' was:

${IPTABLES_OUTPUT}
```

**Review by casperklein — CHANGES_REQUESTED** [review]

File: target/bin/fail2ban (hunk 1 above)
Comment (georglauterbach) [reply target/bin/fail2ban:10]: I didn't even know this was possible 😂👌🏽 Nice!

**Comment by georglauterbach** [comment]
Same as #2160, CI fails unexpectedly....

**Comment by polarathene** [comment]
> CI fails unexpectedly

One of the failing tests is same from https://github.com/docker-mailserver/docker-mailserver/pull/2160#issuecomment-913057642

```
not ok 209 wait_for_empty_mail_queue_in_container fails when timeout reached
# (from function `repeat_until_success_or_timeout' in file test/test_helper/common.bash, line 27,
#  from function `repeat_in_container_until_success_or_timeout' in file test/test_helper/common.bash, line 70,
#  from function `wait_for_empty_mail_queue_in_container' in file test/test_helper/common.bash, line 192,
#  in test file test/test_helper.bats, line 225)
#   `! TEST_TIMEOUT_IN_SECONDS=0 wait_for_empty_mail_queue_in_container "${CONTAINER_NAME}"' failed
# 220 mail.my-domain.com ESMTP
# 250 mail.my-domain.com
# 250 2.1.0 Ok
# 250 2.1.5 Ok
# 354 End data with <CR><LF>.<CR><LF>
# 250 2.0.0 Ok: queued as 85D552F9B0D
# 221 2.0.0 Bye
# efcc772b8339f557c6a12df6038067bac80dcb2fa967904538726d680a946bf4
not ok 210 wait_for_empty_mail_queue_in_container succeeds within timeout
# (in test file test/test_helper.bats, line 250)
#   `[[ ${SECONDS} -gt 0 ]]' failed
# 220 mail.my-domain.com ESMTP
# 250 mail.my-domain.com
# 250 2.1.0 Ok
# 250 2.1.5 Ok
# 354 End data with <CR><LF>.<CR><LF>
# 250 2.0.0 Ok: queued as 8541D2F9B0C
# 221 2.0.0 Bye
# 68b8920e0d61114b05cb90e9b274fb0615f9925b6bec7c8b46cd5c20c2930ae2
```

This one is different however `checking user login: predefined user can login`:

```
not ok 303 checking user login: predefined user can login
# (from function `assert_output' in file test/test_helper/bats-assert/src/assert.bash, line 239,
#  in test file test/tests.bats, line 935)
#   `assert_output "passdb: pass@localhost.localdomain auth succeeded"' failed
# 
# -- output differs --
# expected : passdb: pass@localhost.localdomain auth succeeded
# actual   : passdb: pass@localhost.localdomain auth failed
# --
# 
```

Is that likely F2B related?

File: target/bin/fail2ban [hunk 2]
```diff
@@ -3,6 +3,23 @@
 # shellcheck source=../scripts/helper-functions.sh
 . /usr/local/bin/helper-functions.sh
 
+if ! IPTABLES_OUTPUT=$(iptables -L 2>&1)
+then
+  echo "IPTables does not seem to work properly. The output of 'iptables -L' was:
+
+${IPTABLES_OUTPUT}
+
+Possible causes for this errore are
```
Comment (polarathene) [inline target/bin/fail2ban:12]:
```suggestion
Possible causes for this error are
```

File: target/bin/fail2ban [hunk 3]
```diff
@@ -3,6 +3,23 @@
...
+
+1. Missing capabilities
+2. Modification by user-patches.sh
+
+Aborting.
```
Comment (polarathene) [inline target/bin/fail2ban:17]: Usually 2 or 3 dots for an output implying an action being taken? Either that or no dots.

```suggestion
Aborting..
```

File: target/bin/fail2ban [hunk 4]
```diff
@@ -3,6 +3,23 @@
 # shellcheck source=../scripts/helper-functions.sh
 . /usr/local/bin/helper-functions.sh
 
+if ! IPTABLES_OUTPUT=$(iptables -L 2>&1)
+then
+  echo "IPTables does not seem to work properly. The output of 'iptables -L' was:
```
Comment (polarathene) [inline target/bin/fail2ban:8]:
```suggestion
  echo "IPTables is not functioning correctly. The output of 'iptables -L' was:
```

**Review by polarathene — CHANGES_REQUESTED** [review]

File: target/bin/fail2ban (hunk 2 above)
Comment (georglauterbach) [reply target/bin/fail2ban]: Done. [334898a8a79b16c85984e02d8c5506f5f9792e0c]

File: target/bin/fail2ban (hunk 3 above)
Comment (georglauterbach) [reply target/bin/fail2ban]: Done. [334898a8a79b16c85984e02d8c5506f5f9792e0c]

File: target/bin/fail2ban (hunk 4 above)
Comment (georglauterbach) [reply target/bin/fail2ban]: Done. [334898a8a79b16c85984e02d8c5506f5f9792e0c]

**Comment by georglauterbach** [comment]
We'll see what tests do now. I don't see a connection between the second test and Fail2Ban, and even if there were, we did not change the F2B configuration to begin with :)

**Review by polarathene — APPROVED** [review]

**Review by casperklein — APPROVED** [review]

**Pull request merged**